frontend/src/contexts/agreement.tsx

```diff
@@ -15,13 +16,46 @@ const AgreementProvider: FC = () => {
       agreementId,
     },
   });
+
+  const [voteFn, { error: voteError }] = useMutation(voteMutation, {
+    refetchQueries: ['votes'],
+  });
+
+  const onVote = async (user_id: number, version_id: number, type: 'up' | 'down') => {
```

Comment:
Let's rename this to just "vote", as this isn't an event handler

Reply:
done